core/src/main/java/com/linecorp/armeria/common/MediaTypeSet.java

```diff
@@ -177,23 +197,23 @@ public int size() {
         int matchNumParams = Integer.MIN_VALUE;    // higher = better
         for (MediaType range : ranges) {
             requireNonNull(range, "ranges contains null.");
+            float qValue = range.qualityFactor(Float.NEGATIVE_INFINITY);
```

Comment:
If none of the candidates match, these few lines are wasteful. If all of them do, though it's better to only run them once. I figured it's more likely to match than not match and rearranged this code a little. WDYT?

Reply:
Sounds reasonable.